lib/ansible/plugins/cache/__init__.py

```diff
@@ -154,12 +155,22 @@ def get(self, key):
     def set(self, key, value):
 
         self._cache[key] = value
-
-        cachefile = self._get_cache_file_name(key)
         try:
-            self._dump(value, cachefile)
-        except (OSError, IOError) as e:
-            display.warning("error in '%s' cache plugin while trying to write to %s : %s" % (self.plugin_name, cachefile, to_bytes(e)))
+            cachefile = self._get_cache_file_name(key)
+            tmpfile_handle, tmpfile_path = tempfile.mkstemp(dir=os.path.dirname(cachefile))
```

Comment:
These 2 lines should happen before the `try`, so tmpfile_path is always defined for the `finally`.